lib/SILOptimizer/PassManager/PassPipeline.cpp

```diff
@@ -189,6 +182,19 @@ static void addMandatoryDiagnosticOptPipeline(SILPassPipelinePlan &P) {
   // No uses after consume operator of copyable value.
   P.addConsumeOperatorCopyableValuesChecker();
 
+  // Check ~Escapable.
+  if (P.getOptions().EnableLifetimeDependenceDiagnostics) {
+    P.addLifetimeDependenceDiagnostics();
+  }
+
+  // Devirtualize deinits early if requested.
+  //
+  // FIXME: why is DeinitDevirtualizer in the middle of the mandatory pipeline,
+  // and what passes/compilation modes depend on it? This pass is never executed
+  // or tested without '-Xllvm enable-deinit-devirtualizer'.
```

Comment:
@eeckstein if you know why the DeinitDevirtualizer runs in the middle of the mandatory pipeline or know of any passes that depend on it, in embedded mode or otherwise, please document that in the pass header. I couldn't figure it out after wasting an hour investigating.